packages/lib-sourcify/src/lib/verification.ts

```diff
@@ -273,50 +258,72 @@ export async function verifyCreate2(
+  match.deployedRuntimeBytecode = deployedRuntimeBytecode;
+
   // Check if is a library with call protection
   // See https://docs.soliditylang.org/en/v0.8.19/contracts.html#call-protection-for-libraries
-  recompiledDeployedBytecode = checkCallProtectionAndReplaceAddress(
-    recompiledDeployedBytecode,
-    deployedBytecode
+  recompiledRuntimeBytecode = checkCallProtectionAndReplaceAddress(
+    recompiledRuntimeBytecode,
+    deployedRuntimeBytecode
   );
 
+  if (match.runtimeTransformations === undefined) {
+    match.runtimeTransformations = [];
+  }
+  if (match.runtimeValues === undefined) {
+    match.runtimeValues = {};
+  }
+
   // Replace the library placeholders in the recompiled bytecode with values from the deployed bytecode
   const { replaced, libraryMap } = addLibraryAddresses(
-    recompiledDeployedBytecode,
-    deployedBytecode
+    recompiledRuntimeBytecode,
+    deployedRuntimeBytecode,
+    match.runtimeTransformations
   );
-  recompiledDeployedBytecode = replaced;
+  recompiledRuntimeBytecode = replaced;
+  match.runtimeValues.libraries = libraryMap;
 
   if (immutableReferences) {
-    deployedBytecode = replaceImmutableReferences(
+    deployedRuntimeBytecode = replaceImmutableReferences(
       immutableReferences,
-      deployedBytecode
+      deployedRuntimeBytecode,
+      match.runtimeTransformations
     );
+    match.runtimeValues.immutables == immutableReferences;
   }
 
-  if (recompiledDeployedBytecode === deployedBytecode) {
+  if (recompiledRuntimeBytecode === deployedRuntimeBytecode) {
     match.libraryMap = libraryMap;
     match.immutableReferences = immutableReferences;
     // if the bytecode doesn't contain metadata then "partial" match
-    if (doesContainMetadataHash(deployedBytecode)) {
-      match.status = 'perfect';
+    if (doesContainMetadataHash(deployedRuntimeBytecode)) {
+      match.runtimeMatch = 'perfect';
+      const [, auxdata] = splitAuxdata(deployedRuntimeBytecode);
+      match.runtimeValues.cborAuxdata = { 0: auxdata };
     } else {
-      match.status = 'partial';
+      match.runtimeMatch = 'partial';
     }
   } else {
     // Try to match without the metadata hashes
-    const [trimmedDeployedBytecode] = splitAuxdata(deployedBytecode);
+    const [trimmedDeployedBytecode, auxdata] = splitAuxdata(
+      deployedRuntimeBytecode
+    );
     const [trimmedCompiledRuntimeBytecode] = splitAuxdata(
-      recompiledDeployedBytecode
+      recompiledRuntimeBytecode
     );
     if (trimmedDeployedBytecode === trimmedCompiledRuntimeBytecode) {
       match.libraryMap = libraryMap;
       match.immutableReferences = immutableReferences;
-      match.status = 'partial';
+      match.runtimeMatch = 'partial';
+      match.runtimeTransformations?.push(
+        AuxdataTransformation(trimmedCompiledRuntimeBytecode.length, '0')
+      );
+      match.runtimeValues.cborAuxdata = { '0': auxdata };
     }
   }
```

Comment:
I thought we are using the auxdata extraction algorithm here to match contracts, when there are multiple auxdata

Reply:
We are not, for now we are using this only for the alliance database, I asked you and we agreed to skip this for now. After we merge this we can work to support it in sourcify.